(CDAP-16353) Refactor k8s deployment building in KubeTwillPreparer to a separate function

wyzhang opened the pull request:
What:
Refactor the k8s deployment object building logic into a separate function
with all external data passed in via parameters (thus no access to member fields)

Why:
Allow us to write a unit test for this function to verify the state of
deployment k8s object is as expected.

chtyim commented on cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java:
```diff
@@ -321,7 +322,21 @@ public TwillController start(long timeout, TimeUnit timeoutUnit) {
+   * Return a {@link V1Deployment} object that will be deployed in Kubernetes to run the program
+   */
+  @VisibleForTesting
+  public V1Deployment buildDeployment(RunId runid,
```
Usually method exposed for testing should be kept as package private instead of public.

wyzhang replied:
Fixed

wyzhang committed:
(CDAP-16353) Refactor KubeTwillPreparer k8s deployment building into its own function

What:
Refactor the k8s deployment object building into a separate function
with all external data required passed in via parameters.

Why:
Allow us to write a unit test for this function to verify the state of
deployment k8s object is as expectd.
